### DISCUSSION TITLE
Unity drawbacks

### DISCUSSION THREAD

**Discussion by dmitry-kandiner** [westerndigitalcorporation/stat#36] [Q&A] [answered]
Hello there!

In [6.3. Unity Harness](https://github.com/westerndigitalcorporation/stat#63-unity-harness) of the manual you've mentioned that _Unity_ was the "almost perfect" match. I wonder what were the drawbacks of it, what features it lacked?

Thank you!

**Comment by are-scenic** [comment, marked as answer]
Hi Dmitry,

I didn't imply that Unity has drawbacks. What I meant was that it was lacking certain functionality that was critical for methodology that we used. Unity is a fantastic framework, and it is great when all the system including all teams collaborates in this endeavor. 
Many times, the system is willing to invest in something like this, it requires first to show it brings values. Without collaboration it's hard to achieve. That is why we built around Unity a facility that allows starting with unit-testing and TDD (in particular) in no time even if no one supports the effort.
STAT also provides a Mocking library for test-double that allows constructing test-doubles of many types an variations, much more than just Mock-driven. This flexibility was needed to make it simpler to build test-doubles that suit better concrete needs of specific use-cases, in which using pure Mocks is not very convenient. Mocking library of STAT is takes many ideas from _mock_ library of _unittest_ module in Python.

I hope it answers your question.

Kind Regards,
Arseniy

**Reply by dmitry-kandiner** [reply]
Thank you very much, it definitely does :)
